tests/lib/sample_rate_converter/src/main.c

```diff
@@ -0,0 +1,1065 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/tc_util.h>
+#include <sample_rate_converter.h>
+
+struct sample_rate_converter_ctx conv_ctx;
+
+static void test_setup(void *f)
+{
+	sample_rate_converter_open(&conv_ctx);
+}
+
+#ifdef CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_16
+ZTEST(suite_sample_rate_converter, test_init_valid_decimate_24khz_16bit)
+{
+	int ret;
+
+	uint32_t input_sample_rate = 48000;
+	uint32_t output_sample_rate = 24000;
+	uint32_t conversion_ratio = input_sample_rate / output_sample_rate;
+
+	uint16_t input_samples[] = {1000, 2000, 3000, 4000,  5000,  6000,
+				    7000, 8000, 9000, 10000, 11000, 12000};
+	size_t num_samples = ARRAY_SIZE(input_samples);
+	size_t expected_output_samples = num_samples / conversion_ratio;
+	uint16_t output_samples[expected_output_samples];
+
+	enum sample_rate_converter_filter filter = SAMPLE_RATE_FILTER_SIMPLE;
+	size_t output_written;
+
+	ret = sample_rate_converter_process(
+		&conv_ctx, filter, input_samples, num_samples * sizeof(uint16_t), input_sample_rate,
+		output_samples, expected_output_samples * sizeof(uint16_t), &output_written,
+		output_sample_rate);
+
+	zassert_equal(ret, 0, "Sample rate conversion process failed");
+	zassert_equal(conv_ctx.input_sample_rate, input_sample_rate,
+		      "Input sample rate not as expected");
+	zassert_equal(conv_ctx.output_sample_rate, output_sample_rate,
+		      "Output sample rate not as expected");
+	zassert_equal(conv_ctx.conversion_ratio, conversion_ratio,
+		      "Conversion ratio not as expected");
+	zassert_equal(conv_ctx.conversion_direction, CONVERSION_DIR_DOWN,
+		      "Conversion direction is not as expected");
+	zassert_equal(conv_ctx.filter_type, filter, "Filter not as expected");
+	zassert_equal(conv_ctx.input_buf.bytes_in_buf, 0, "Bytes in input buffer not as expected");
+	zassert_equal(ring_buf_size_get(&conv_ctx.output_ringbuf), 0,
+		      "Number of bytes in output ringbuffer not as expected %d",
+		      ring_buf_size_get(&conv_ctx.output_ringbuf));
+
+	zassert_equal(output_written, expected_output_samples * sizeof(uint16_t),
+		      "Output size was not as expected (%d)", output_written);
+
+	/* Verify output bytes, expected two input samples per output sample */
+	for (int i = 0; i < expected_output_samples; i++) {
+		zassert_within(output_samples[i], input_samples[i * 2], 550);
```

Comment:
We need to define 550 and provide some reasoning behind this value.

Reply:
I agree, they are way to arbitrary and magic as they are now - but I'm not sure how to find a good, scientific reasoning for it. Can we go through the tests offline and do some brainstorming for it?